monitoring/monitoring.go

```diff
@@ -231,8 +231,14 @@ func (d *derivedFloat64Metric) Register() error {
 type float64Metric struct {
 	*stats.Float64Measure
 
+	// tags stores all tags for the metrics
 	tags []tag.Mutator
+	// ctx is a precomputed context holding tags, as an optimization
+	ctx  context.Context
```

Comment:
This technically violates the explicit documentation for `context.Context`:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

But I'm not sure that matters for this use case. This is a vestige of the OC library construction, and is not holding request-scoped information or any sort of deadlines, etc.